src/coreclr/jit/importer.cpp

```diff
@@ -5625,7 +5626,7 @@ GenTree* Compiler::impCastClassOrIsInstToTree(
     if (isCastClass)
     {
         assert((helper == CORINFO_HELP_CHKCASTCLASS) || (helper == CORINFO_HELP_CHKCASTARRAY) ||
-               (helper == CORINFO_HELP_CHKCASTINTERFACE));
+               (helper == CORINFO_HELP_CHKCASTANY) || (helper == CORINFO_HELP_CHKCASTINTERFACE));
```

Comment:
Is sounds like a bug that we can end up here for `CORINFO_HELP_CHKCASTINTERFACE`. It does not make sense to inline the MT check for interface casts - it is always going to be false.

Reply:
It is used for profiled casts, e.g.:

```csharp
static IDisposable CastInterface(object o)
{
    return (IDisposable)o;
}
```
emits:
```asm
       sub      rsp, 40
       mov      rax, rcx
       test     rax, rax
       je       SHORT G_M63531_IG05
       mov      rdx, 0x7FFBDF89AE08      ; MyDisposable (class implementing IDisposable)
       cmp      qword ptr [rax], rdx
       je       SHORT G_M63531_IG05
       mov      rdx, rcx
       mov      rcx, 0x7FFBDEFF1C30      ; System.IDisposable
       call     [CORINFO_HELP_CHKCASTINTERFACE]
G_M63531_IG05:  
       nop      
       add      rsp, 40
       ret      
; Total bytes of code 52
```
Cast profiling is only enabled for static pgo training (`DOTNET_JitProfileCasts=1`)